.github/workflows/workflow.yml

```
@@ -172,9 +178,24 @@ jobs:
         with:
           path: ${{ env.pythonLocation }}
           key: ${{ env.pythonLocation }}-build-${{ hashFiles('setup.py') }}
+      - name: Cache build release
+        id: restore-build-release
+        uses: actions/cache@v2
+        with:
+          path: dist
+          key: ${{ env.pythonLocation }}-build-release-${{ hashFiles('setup.py') }}
       - name: Upload a Python package to PyPi
         run: twine upload dist/* --username $PYPI_USERNAME --password $PYPI_PASSWORD
       - name: Publish a git tag
         run: "${GITHUB_WORKSPACE}/.github/publish-git-tag.sh"
-      - name: Deploy the Web API
-        run: ssh -o StrictHostKeyChecking=no deploy-api@fr.openfisca.org
+      - name: Add SSH key & Deploy the Web API
+        env:
+          SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+        run: |
+          mkdir -p /home/runner/.ssh
+          ssh-keyscan fr.openfisca.org >> /home/runner/.ssh/known_hosts
+          echo "${{ secrets.FRANCE_API_DEPLOY_KEY }}" > /home/runner/.ssh/github_actions
```

This key is more related to a specific deployment than to the user 🙂
```suggestion
          echo "${{ secrets.FRANCE_API_DEPLOY_KEY }}" > /home/runner/.ssh/france_api_deploy
```